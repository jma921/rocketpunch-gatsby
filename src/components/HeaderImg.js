import React from 'react';
import Img from 'gatsby-image';
import styled from 'styled-components';
import PropTypes from 'prop-types';

// const Image = styled(Img)`
//   margin-top: -52px;
//   display: inherit;
//   height: 65vh;
// `;

const Image = styled(Img)`
  display: block;
  margin-top: -52px;
  position: inherit;
  width: 100%;
  height: 65vh;
  cursor: pointer;
  .gatsby-image-wrapper > img {
    position: inherit;
  }
`;

const Overlay = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 65vh;
  width: 80%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
`;

const Text = styled.p`
  color: white;
  text-align: left;
  font-size: 2rem;
  font-weight: normal;
`;

const Span = styled.span`
  background: white;
  color: #ff4401;
  padding: 0.25rem;
  font-weight: 600;
`;

const HeaderImg = props => {
  return (
    <div>
      <Image sizes={props.sizes} />
      <Overlay>
        <Text>
          We create beautifully-crafted designs that <Span>stand out</Span> from
          the crowd.
        </Text>
      </Overlay>
    </div>
  );
};

HeaderImg.propTypes = {
  sizes: PropTypes.object
};

export default HeaderImg;
