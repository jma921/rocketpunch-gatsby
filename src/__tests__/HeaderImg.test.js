import React from 'react';
import renderer from 'react-test-renderer';
import HeaderImg from '../components/HeaderImg';

const sizes = {
  aspectRatio: 1.4109347442680775,
  base64:
    'data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAOABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAIEAwf/xAAgEAABBAIBBQAAAAAAAAAAAAABAAIDIQQSMREiQYHR/8QAFgEBAQEAAAAAAAAAAAAAAAAABAMF/8QAGxEAAgMAAwAAAAAAAAAAAAAAAQIAEjEDBBH/2gAMAwEAAhEDEQA/AONAdfFDlaBoTtio9x+pNSJNSUS3uTTqV0S6GFjomkyRtrgoUWZWQ7WhVekKI4WYWtsQewqGpXJ//9k=',
  sizes: '(max-width: 1080px) 100vw, 1080px',
  src: '/static/dark-bg-1eba63084a303b30ed3ba9939c27ee26-978f9.png'
};

test('Renders HeaderImg component', () => {
  const component = renderer.create(<HeaderImg sizes={sizes} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  //   tree.props.onMouseEnter();
  //   // re-rendering
  //   tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();

  //   // manually trigger the callback
  //   tree.props.onMouseLeave();
  //   // re-rendering
  //   tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
});
