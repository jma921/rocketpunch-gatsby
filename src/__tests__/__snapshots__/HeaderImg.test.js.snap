// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders HeaderImg component 1`] = `
<div>
  <div
    className=" gatsby-image-outer-wrapper"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="sc-bdVaJa gpwlEx gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "70.875%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        onLoad={undefined}
        src="data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAOABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAIEAwf/xAAgEAABBAIBBQAAAAAAAAAAAAABAAIDIQQSMREiQYHR/8QAFgEBAQEAAAAAAAAAAAAAAAAABAMF/8QAGxEAAgMAAwAAAAAAAAAAAAAAAQIAEjEDBBH/2gAMAwEAAhEDEQA/AONAdfFDlaBoTtio9x+pNSJNSUS3uTTqV0S6GFjomkyRtrgoUWZWQ7WhVekKI4WYWtsQewqGpXJ//9k="
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 0.5s",
            "transitionDelay": "0.25s",
            "width": "100%",
          }
        }
        title={undefined}
      />
      <img
        alt=""
        onLoad={[Function]}
        sizes="(max-width: 1080px) 100vw, 1080px"
        src="/static/dark-bg-1eba63084a303b30ed3ba9939c27ee26-978f9.png"
        srcSet={undefined}
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 0.5s",
            "width": "100%",
          }
        }
        title={undefined}
      />
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<img width=\\"\\" height=\\"\\" src=\\"/static/dark-bg-1eba63084a303b30ed3ba9939c27ee26-978f9.png\\" srcset=\\"undefined\\" alt=\\"\\" title=\\"\\" sizes=\\"(max-width: 1080px) 100vw, 1080px\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/>",
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-bwzfXH bZgOVJ"
  >
    <p
      className="sc-htpNat jYQgjw"
    >
      We create beautifully-crafted designs that 
      <span
        className="sc-bxivhb hdvesG"
      >
        stand out
      </span>
       from the crowd.
    </p>
  </div>
</div>
`;
